crates/libs/windows/src/Windows/UI/UIAutomation/Core/impl.rs

```diff
@@ -1,5 +1,5 @@
 pub trait ICoreAutomationConnectionBoundObjectProvider_Impl: Sized {
-    fn IsComThreadingRequired(&self) -> windows_core::Result<bool>;
+    fn IsComThreadingRequired(&self) -> bool;
```

Comment:
There are a handful of system APIs that adopted the `noexcept` attribute. It really should be applied more liberally and can be done retroactively to improve code size for callers.

Reply:
@jonwis 👋

Reply:
Is adding the `noexcept` attribute meant to be a semver-breaking change? With this PR, it seems like it would become one, but I'm not sure if that's desired/acceptable.

Reply:
The crate is pre-release/0.x so there is no _semver-breaking_ change here right?

Reply:
Technically, but because of the Rust community's reluctance to ever release 1.0 crates, cargo treats the minor version of a 0.x crate more like the major version of a 1.x crate.

I'm more worried about the long term. I interpreted Kenny's comment to mean we should update the winmd source files to add `noexcept` where we can, and the part about doing it retroactively means if we start doing that after this crate gets to 1.0 each of those would be a breaking change.

I guess this comes down to what the winmd files consider to be breaking changes. Obviously a breaking change there would imply breaking changes for any downstream projections.

Reply:
Yes, that's a good point. It doesn't break the ABI by adding it, but it would be as if the API changed for Rust callers.

Reply:
@oldnewthing as we just had an internal conversation about noexcept at the _SDK ABI_ layer. Projections, no problem with them adding noexcept decorations to enforce that. Generally the Win32 ABI is already implicitly noexcept since we can't throw you a C++ exception from within combase.dll to your caller and have it make any sense.

Adding the annotation is a similar conversation as nullable from a while ago. Knowing when to apply it automatically is hard but you can probably figure out some basic rules.

But, per our API design guidelines (man I should publish those someday) property fetches & writes are not supposed to raise exceptions; only method calls.